src/server/index.ts

```diff
@@ -254,9 +317,42 @@ export class DocumentDriveServer implements IDocumentDriveServer {
                 throw new Error('Invalid Document Drive document');
             }
 
+            // filter out signals that created a document that was deleted by a subsequent signal
+            const signalResults = signals
+                .map(signal => signal.signal)
+                .filter(
```

Comment:
I added this filter for now to avoid this error but this errors feels like an indication we need a better approach